### ISSUE TITLE
PortraitFrameMixin:SetPortraitToClassIcon errors when setting a portrait

### ISSUE THREAD

**Issue by Meorawr** [Stanzilla/WoWUIBugs#394]
The [PortraitFrameMixin:SetPortraitToClassIcon] method runs into a script error when called as it attempts to pass an asset that isn't 64x64 pixels in size to the [SetPortraitToTexture] API.

In terms of the default UI, this method is only called by the [ClassTalentFrameMixin:UpdatePortrait] if no specialization information available for the player. This can occur if the player loads an addon with a dependency upon Blizzard_ClassTalentUI which triggers a portrait update on-load of the class talent frame.

# Test case

Start the client and run the following command:

```
/run CharacterFrame:SetPortraitToClassIcon("WARRIOR")
```

You'll receive the following script error. Note that this script error may only occur once per client startup - seemingly only on the initial attempt to portrait-ize the "UI-Classes-Circles" texture. Future attempts to use the same portrait again don't run into the error until the client has been restarted, and will instead display an invalid/transparent texture.

```
Message: Interface/SharedXML/PortraitFrame.lua:73: SetPortraitToTexture(): Texture is not 64x64 pixels: Interface/TargetingFrame/UI-Classes-Circles.

Time: Sun Feb  5 11:51:14 2023
Count: 1
Stack: Interface/SharedXML/PortraitFrame.lua:73: SetPortraitToTexture(): Texture is not 64x64 pixels: Interface/TargetingFrame/UI-Classes-Circles.

[string "=[C]"]: in function `SetPortraitToTexture'
[string "@Interface/SharedXML/PortraitFrame.lua"]:73: in function `SetPortraitToAsset'
[string "@Interface/SharedXML/PortraitFrame.lua"]:93: in function `SetPortraitToClassIcon'
[string "CharacterFrame:SetPortraitToClassIcon("WARRIOR")"]:1: in main chunk
[string "=[C]"]: in function `RunScript'
[string "@Interface/FrameXML/ChatFrame.lua"]:2175: in function `?'
[string "@Interface/FrameXML/ChatFrame.lua"]:5229: in function `ChatEdit_ParseText'
[string "@Interface/FrameXML/ChatFrame.lua"]:4893: in function `ChatEdit_SendText'
[string "@Interface/FrameXML/ChatFrame.lua"]:4929: in function `ChatEdit_OnEnterPressed'
[string "*ChatFrame.xml:127_OnEnterPressed"]:1: in function <[string "*ChatFrame.xml:127_OnEnterPressed"]:1>

Locals: (*temporary) = CharacterFramePortrait {
 0 = <userdata>
}
(*temporary) = "Interface/TargetingFrame/UI-Classes-Circles"
```

# Solutions

Replacing the `SetPortraitToAsset` call with one to `SetPortraitTextureRaw` resolves the issue and should work well enough as the individual class icons are already pre-baked with a circular mask.

```diff
diff --git a/Interface/SharedXML/PortraitFrame.lua b/Interface/SharedXML/PortraitFrame.lua
index 7e98b74f..bb0a2c3b 100644
--- a/Interface/SharedXML/PortraitFrame.lua
+++ b/Interface/SharedXML/PortraitFrame.lua
@@ -90,7 +90,7 @@ function PortraitFrameMixin:SetPortraitAtlasRaw(atlas, ...)
 end
 
 function PortraitFrameMixin:SetPortraitToClassIcon(classFilename)
-	self:SetPortraitToAsset("Interface/TargetingFrame/UI-Classes-Circles");
+	self:SetPortraitTextureRaw("Interface/TargetingFrame/UI-Classes-Circles");
 	local left, right, bottom, top = unpack(CLASS_ICON_TCOORDS[string.upper(classFilename)]);
 	self:SetPortraitTexCoord(left, right, bottom, top);
 end
```

[PortraitFrameMixin:SetPortraitToClassIcon]: https://github.com/Gethe/wow-ui-source/blob/72927e493cda0de693929ac194517ff116e286b0/Interface/SharedXML/PortraitFrame.lua#L92-L93
[ClassTalentFrameMixin:UpdatePortrait]: https://github.com/Gethe/wow-ui-source/blob/72927e493cda0de693929ac194517ff116e286b0/Interface/AddOns/Blizzard_ClassTalentUI/Blizzard_ClassTalentFrameTemplate.lua#L155-L158
[SetPortraitToTexture]: https://wowpedia.fandom.com/wiki/API_SetPortraitToTexture

**Comment by JourneyOver** [comment]
Yep been running into the same issue myself the past couple days or so now or at least I think it is the same issue.. Only happens on initial load of a toon, but any subsequent /reload or moving between zones does not continue to produce the error. lua error below:

```lua
Interface/SharedXML/PortraitFrame.lua:73: SetPortraitToTexture(): Texture is not 64x64 pixels: Interface/TargetingFrame/UI-Classes-Circles.

Count: 1

Call Stack:
[string "=[C]"]: ?
[string "=[C]"]: in function `SetPortraitToTexture'
[string "@Interface/SharedXML/PortraitFrame.lua"]:73: in function `SetPortraitToAsset'
[string "@Interface/SharedXML/PortraitFrame.lua"]:93: in function `SetPortraitToClassIcon'
[string "@Interface/AddOns/Blizzard_ClassTalentUI/Blizzard_ClassTalentFrameTemplate.lua"]:158: in function `UpdatePortrait'
[string "@Interface/AddOns/Blizzard_ClassTalentUI/Blizzard_ClassTalentFrameTemplate.lua"]:21: in function <..._ClassTalentUI/Blizzard_ClassTalentFrameTemplate.lua:11>
```

**Comment by Meorawr** [comment]
Fixed in 11.0.0.
